src/analysis/bounds.rs

```diff
@@ -107,7 +107,11 @@ impl Bounds {
                 ));
                 if r#async && (par.name == "callback" || par.name.ends_with("_callback")) {
                     let func_name = func.c_identifier.as_ref().unwrap();
-                    let finish_func_name = finish_function_name(func_name);
+                    let finish_func_name = if let Some(finish_func_name) = &func.finish_func {
```

Comment:
In addition, if there is a finish function you can know that the function is async and get around the "is this function async" heuristics

Reply:
Done in a new commit.